packages/react-sdk/src/core/hooks/useCalculateHardLimit.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from 'react';
+
+export const useCalculateHardLimit = (
+  /**
+   * Element that stretches to 100% of the whole layout component
+   */
+  wrapperElement: HTMLDivElement | null,
+  /**
+   * Element that directly hosts individual `ParticipantView` (or wrapper) elements
+   */
+  hostElement: HTMLDivElement | null,
+  limit?: 'dynamic' | number,
+) => {
+  const [calculatedLimit, setCalculatedLimit] = useState<{
+    vertical: number;
+    horizontal: number;
+  }>({
+    vertical: typeof limit === 'number' ? limit : 1,
+    horizontal: typeof limit === 'number' ? limit : 1,
+  });
+
+  useEffect(() => {
+    if (
+      !hostElement ||
+      !wrapperElement ||
+      typeof limit === 'number' ||
+      typeof limit === 'undefined'
+    )
+      return;
+
+    let childWidth = 280;
+    let childHeight = 160;
```

Comment:
Leftovers from testing?

Reply:
Defaults to allow at least one element to render to get proper values for future calculations but now looking at it I'd rather do `calculatedLimit || 1` in the return statement.

Reply:
Resolved here: https://github.com/GetStream/stream-video-js/pull/1090/commits/6537678ace8f208f30acce565cc1cfe4b76be864